### Load packages and dataset

In [1]:
# import packages

import pandas as pd
import numpy as np
from matplotlib import pyplot as plt
import math

In [2]:
# load dataset

bmx_data = pd.read_csv("datasets/BMX_data.csv")
bmx_data.head()

SEQN  BMDSTATS  BMXWT  BMIWT  BMXRECUM  BMIRECUM  BMXHEAD  BMIHEAD  BMXHT  \
0  62161         1   69.2    NaN       NaN       NaN      NaN      NaN  172.3   
1  62162         1   12.7    NaN      95.7       NaN      NaN      NaN   94.7   
2  62163         1   49.4    NaN       NaN       NaN      NaN      NaN  168.9   
3  62164         1   67.2    NaN       NaN       NaN      NaN      NaN  170.1   
4  62165         1   69.1    NaN       NaN       NaN      NaN      NaN  159.4   

   BMIHT    ...     BMXARMC  BMIARMC  BMXWAIST  BMIWAIST  BMXSAD1  BMXSAD2  \
0    NaN    ...        32.5      NaN      81.0       NaN     17.7     17.9   
1    NaN    ...        16.6      NaN      45.4       NaN      NaN      NaN   
2    NaN    ...        22.0      NaN      64.6       NaN     15.6     15.5   
3    NaN    ...        29.3      NaN      80.1       NaN     18.3     18.5   
4    NaN    ...        29.7      NaN      86.7       NaN     21.0     20.8   

   BMXSAD3  BMXSAD4  BMDAVSAD  BMDSADCM  
0      NaN      NaN      17.8       NaN  
1      NaN      NaN       NaN       NaN  
2      NaN      NaN      15.6       NaN  
3      NaN      NaN      18.4       NaN  
4      NaN      NaN      20.9       NaN  

[5 rows x 26 columns]

## Task 1: Compute a single 95% confidence interval

In [3]:
# calculate "population" mean

all_weights = bmx_data["BMXWT"] 
pop_weight_mean = np.mean(all_weights)
print(pop_weight_mean)

61.52891918208375


<p>
<details><summary>Click to show answer</summary><br>
    
`61.52891918208375`

</details>
</p>

<p>
<details><summary>Click to show solution</summary><br>
    
```python
# calculate "population" mean

all_weights = bmx_data["BMXWT"]
pop_weight_mean = np.mean(all_weights)
print(pop_weight_mean)
```

</details>
</p>

In [4]:
# create sample of 1000 weight values

sample_weights = bmx_data["BMXWT"].sample(1000)

print(list(sample_weights[:10])) #print the first 10 values in this sample

[110.1, 102.8, 57.6, 103.9, 75.4, 69.2, 118.0, 81.4, 41.4, 9.9]


In [5]:
# calculate sample mean

sample_mean = np.mean(sample_weights)
print(sample_mean)

60.54068825910932


<p>
<details><summary>Click to show solution</summary><br>
    
```python
# calculate sample mean

sample_mean = np.mean(sample_weights)
print(sample_mean)
```

</details>
</p>

In [6]:
# calculate sample standard deviation

sample_stdev = np.std(sample_weights)
print(sample_stdev)

31.17981940274571


<p>
<details><summary>Click to show solution</summary><br>
    
```python
# calculate sample standard deviation

sample_stdev = np.std(sample_weights)
print(sample_stdev)
```

</details>
</p>

In [ ]:
# calculate 95% confidence interval for weight

n = 1000 #sample size




print(ci)

<p>
<details><summary>Click to show solution</summary><br>
    
```python
# calculate 95% confidence interval for weight

n = 1000 #sample size

low_ci_bound = sample_mean - 1.96 * (sample_stdev/math.sqrt(n))
high_ci_bound = sample_mean + 1.96 * (sample_stdev/math.sqrt(n))
ci = [low_ci_bound, high_ci_bound]
print(ci)
```

</details>
</p>